### PR TITLE
build: upgrade FOCA & fix ReadTheDocs

### PR DESCRIPTION
**Pull request by alohamora**
## Description

1. Upgrade foca to the latest version (0.12.1). This change introduces no updates in config/functionality as the foca upgrade doesn't introduce any breaking changes.
2. fix readthedocs build failure by adding documentation build config and requirements

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist:

- [ ] My code follows the [style guidelines](https://github.com/elixir-cloud-aai/elixir-cloud-aai/blob/dev/resources/contributing_guidelines.md#language-specific-guidelines) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have not reduced the existing code coverage
- [ ] I have added docstrings following the [Python style guidelines](https://github.com/elixir-cloud-aai/elixir-cloud-aai/blob/dev/resources/python.md) of this project to all new modules, classes, methods and functions are documented with docstrings following; I have updated any previously existing docstrings, if applicable
- [ ] I have updated any sections of the app's documentation that are affected by the proposed changes, if applicable

### CODE REVIEW

**Comment by alohamora** [comment]
@uniqueg the RtD ci step still failed, as per the error, it is trying to find a file named `.readthedocs.yaml` for it's configuration which does not exist in our repo

**Comment by alohamora** [comment]
finally, all test passed 😛 
had to add additional requirements for building documentation with the sphinx_rtd_theme as now readthedocs does not download this theme package by default.

File: docs/api/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Sphinx==7.2.6
```
Comment (alohamora) [inline docs/api/requirements.txt:1]: all the versions mentioned here are the latest LTS versions of these packages

**Comment by alohamora** [comment]
@uniqueg can you please review this PR

**Review by uniqueg — APPROVED** [review]

**Pull request merged**